cmd/determinize-ci-operator/main.go

```diff
@@ -180,10 +189,72 @@ func migrateOpenshiftInstallerCustomTestImageTemplates(
 	return migratedCount
 }
 
-func ipiWorkflowForClusterProfile(clusterProfile api.ClusterProfile) string {
-	suffix := string(clusterProfile)
+func providerNameForProfile(clusterProfile api.ClusterProfile) string {
 	if clusterProfile == api.ClusterProfileAzure4 {
-		suffix = "azure"
+		return "azure"
+	}
+	return string(clusterProfile)
+}
+
+func ipiWorkflowForClusterProfile(clusterProfile api.ClusterProfile) string {
+	return fmt.Sprintf("ipi-%s", providerNameForProfile(clusterProfile))
+}
+
+func migrateOpenshiftOpenshiftInstallerUPIClusterTestConfiguration(
+	configuration *config.DataWithInfo,
+	allowedBranches sets.String,
+	allowedOrgs sets.String,
+	allowedCloudproviders sets.String,
+) (migratedCount int, err error) {
+	if (len(allowedBranches) != 0 && !allowedBranches.Has(configuration.Info.Branch)) || (len(allowedOrgs) != 0 && !allowedOrgs.Has(configuration.Info.Org)) {
+		return 0, nil
+	}
+
+	for idx, test := range configuration.Configuration.Tests {
+		if test.OpenshiftInstallerUPIClusterTestConfiguration == nil ||
+			(len(allowedCloudproviders) != 0 && !allowedCloudproviders.Has(string(test.OpenshiftInstallerUPIClusterTestConfiguration.ClusterProfile))) {
+			continue
+		}
+
+		commandFields := strings.Fields(test.Commands)
+		if n := len(commandFields); n != 2 {
+			return 0, fmt.Errorf("command %q didn't have exactly two fields", test.Commands)
```

Comment:
I don't think this needs to be error-level. If we have strange tests like this, I think it's fine to just skip them for now, maybe warn about them.

Reply:
Updated